saneterm/pty.py

```diff
@@ -94,22 +361,196 @@ def parse(self, input):
+                                        elif sgr_type == 48:
+                                            try:
+                                                self.__text_style.bg_color = parse_extended_color(arg_it)
+                                            except AssertionError:
+                                                this_supported = False
+                                        elif sgr_type == 49:
+                                            self.__text_style.bg_color = None
+                                        elif sgr_type >= 90 and sgr_type <= 97:
+                                            self.__text_style.fg_color = Color(
+                                                ColorType.NUMBERED_8_BRIGHT,
+                                                BasicColor(sgr_type - 90)
+                                            )
+                                        elif sgr_type >= 100 and sgr_type <= 107:
+                                            self.__text_style.bg_color = Color(
+                                                ColorType.NUMBERED_8_BRIGHT,
+                                                BasicColor(sgr_type - 100)
+                                            )
+                                        else:
+                                            # Not supported:
+                                            #   5-6     blink
+                                            #   7       invert
+                                            #   10      default font
+                                            #   11-19   alternative font
+                                            #   20      blackletter font
+                                            #   25      disable blinking
+                                            #   26      proportional spacing
+                                            #   27      disable inversion
+                                            #   50      disable proportional spacing
+                                            #   51      framed
+                                            #   52      encircled
+                                            #   53      overlined (TODO: implement via GTK 4 TextTag)
+                                            #   54      neither framed nor encircled
+                                            #   55      not overlined
+                                            #   60-65   ideograms (TODO: find out what this is supposed to do)
+                                            #   58-59   underline color, non-standard
+                                            #   73-65   sub/superscript, non-standard (TODO: via scale and rise)
+                                            this_supported = False
+
+                                        supported = supported or this_supported
+
+                                    if supported:
+                                        special_ev = (EventType.TEXT_STYLE, self.__text_style)
+
```

Comment:
Maybe refactor this into a `parse_parameter` function or something along those lines? Seems a bit deeply intended otherwise. Or move the entire code for handling `\033` to a separate `parse_esc` function?

Reply:
Done!